.github/workflows/generate_new_client.yaml

```diff
@@ -0,0 +1,138 @@
+          --product-docs=\"${PRODUCT_DOCS}\" \
+          --api-description=\"${API_DESCRIPTION}\" \
+          --googleapis-gen-url=\"${GOOGLEAPIS_GEN_URL}\""
+
+          # helper function that appends a python argument only if specified in the GH action inputs
+          append_argument() {
+            py_arg=$1
+            # env vars look exactly like new-client arguments but uppercase + underscores
+            env_name=$(echo "${py_arg}" | sed 's/-/_/g' | sed -e 's/\([a-z]\)/\U\1/g')
+            if [[ -n "${!env_name}" ]]; then
+              # $(echo) is redundant but it works around a syntax highlighting problem in vim
+              arguments=$(echo "${arguments} --${py_arg}=\"${!env_name}\"")
+            fi
+          }
+
+          declare -a optional_args=('transport' 'cloud-api' 'group-id' 'rest-docs' 'rpc-docs')
+
+          for python_argument in "${optional_args[@]}"; do
+            append_argument "${python_argument}"
+          done
+          echo "::set-output name=new_library_args::${arguments}"
+          echo "${arguments}" | xargs python generation/new_client/new-client.py generate
+        env:
+          GOOGLEAPIS_GEN_URL: https://yoshi-approver:${{ secrets.YOSHI_APPROVER_PRIVATE_TOKEN }}@github.com/googleapis/googleapis-gen.git
+          API_SHORTNAME: ${{ github.event.inputs.api_shortname }}
+          NAME_PRETTY: ${{ github.event.inputs.name_pretty }}
+          PROTO_PATH: ${{ github.event.inputs.proto_path }}
+          PRODUCT_DOCS: ${{ github.event.inputs.product_docs }}
+          REST_DOCS: ${{ github.event.inputs.rest_docs }}
+          RPC_DOCS: ${{ github.event.inputs.rpc_docs }}
+          API_DESCRIPTION: ${{ github.event.inputs.api_description }}
+          TRANSPORT: ${{ github.event.inputs.transport }}
+          CLOUD_API: ${{ github.event.inputs.CLOUD_API }}
+          GROUP_ID: ${{ github.event.inputs.group_id }}
+      - name: Push to branch
+        run: |
+          [ -z "`git config user.email`" ] && git config --global user.email "${USERNAME:-script}@google.com"
+          [ -z "`git config user.name`" ] && git config --global user.name "${USERNAME:-script}"
+
+          # create and push to branch in origin
+          branch_name="new-library/${{ github.event.inputs.api_shortname }}"
+          git checkout -b "${branch_name}"
+          git add --all
+          git commit -m "feat: [${API_SHORTNAME}] new module for ${API_SHORT_NAME}\n\n\
+          \`\`\`${GENERATION_ARGUMENTS}\`\`\`"
+          git remote add monorepo https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git
+          git fetch -q --unshallow monorepo
+          git push -f monorepo "${branch_name}"
+
+          # create PR
+          gh pr create --fill
+        env:
+          API_SHORTNAME: ${{ github.event.inputs.api_shortname }}
+          GENERATION_ARGUMENTS: ${{ steps.generation.outputs.new_library_args }}
+          GH_TOKEN: ${{ secrets.YOSHI_APPROVER_PRIVATE_TOKEN }}

```

Comment:
I remember we use YOSHI_CODE_BOT_TOKEN for creating pull requests.

Reply:
I switched to YOSHI_CODE_BOT_TOKEN